source-firestore/discovery.go

```diff
@@ -268,25 +277,42 @@ func (ds *discoveryState) processDocument(ctx context.Context, doc *firestore.Do
 				return fmt.Errorf("error listing subcollections: %w", err)
 			}
 			for _, subcoll := range subcolls {
-				ds.processCollection(ctx, subcoll)
+				ds.handleCollection(ctx, subcoll)
 			}
 			return nil
 		})
 	}
 	return nil
 }
 
+func (ds *discoveryState) handleCollection(ctx context.Context, coll *firestore.CollectionRef) {
+	ds.shared.Lock()
```

Comment:
Is this lock needed?

Reply:
You're right, that's not necessary any more. (It's left over from the previous discovery implementation where this function was checking a shared map to make sure it only spawned one worker thread per collection group)